src/gc-stacks.c

```diff
@@ -5,7 +5,25 @@
 #  include <sys/resource.h>
 #endif
 
-const size_t jl_guard_size = (4096 * 16);
+#ifdef _P64
+# ifdef _OS_WINDOWS_
+#  define MAX_STACK_MAPPINGS 500
+# else
+#  define MAX_STACK_MAPPINGS 30000
+# endif
+#else
+# ifdef _OS_WINDOWS_
+#  define MAX_STACK_MAPPINGS 250
+# else
+#  define MAX_STACK_MAPPINGS 500
+# endif
+#endif
```

Comment:
we should probably make these the default values for a runtime variable, since several of these are dependent on the amount of memory available in the system (currently these are set for the size of our CI machines, but user machines may be much larger or smaller), so it may make sense for some users to need to configure these